docs/website/configuration.rst

```diff
@@ -0,0 +1,120 @@
+
+To accomplish this task, we recommend you to use the instance ``g4dn.xlarge`` on Amazon EC2. For convenience, you can search *deep learning* in the filter bar to select the deep learning-enabled machines, where you will have no need of installing addition drivers.
+
+After you have got the machine and logged to the machine, you will see that on the welcome screen, it shows the conda environments if you have selected a deep learning machine. Here we will choose the MXNet one, use the command below:
+
+.. code:: bash
+
+    source activate mxnet_p36
+
+We will use the commands below to update the packages on this machine
+
+.. code:: bash
+
+    sudo apt-get update
+    sudo apt-get upgrade
+
+Use the commands below to install or update necessaryy packages:
+
+.. code:: bash
+
+    pip install notedown
+    pip install --upgrade sphinx
+    pip install --user -U numpy
+    pip install "nltk==3.2.5"
+
+After installing the packages above, get the GluonNLP repository from GitHub via the command:
+
+.. code:: bash
+
+    git clone https://github.com/dmlc/gluon-nlp
+
+Then navigate to the gluon-nlp folder
+
+.. code:: bash
+
+    cd gluon-nlp
+
+Use the command below to install the necessary packages in python:
+
+.. code:: bash
+
+    pip install --user -e '.[dev]'
+
+If necessary, you might still need to configure the packages like below:
+
+Use ``python`` command to get into the python execution screen, and then type the commands below to install the necessary packages inside python:
+
+.. code:: bash
+
+    import nltk
+    nltk.download('perluniprops')
+    nltk.download('nonbreaking_prefixes')
+    nltk.download('punkt')
+
+After all these configuration, you will be able to make the conversion from ``md`` files to the ``ipynb`` files.
+
+Use the command like the one below to do the conversion:
+
+
+.. code:: bash
+
+    python3 docs/md2ipynb.py docs/examples/language_model/language_model.md
+
+And then you will be able to see the result file in the same path as the ``md`` file.
+
+2. Using make docs to convert static files into HTML
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This work can be done on any machine as it doesn't need running the python scripts.
+
+*Requirements*: You have the ``ipynb`` files ready. If you modified the ``md`` files, then you will need the first step to compile the new ``ipynb`` files.
+
+It is also possible that you will need to install some necessary packages to help the ``make docs`` work. This include: python, pip and some other packages which will be installed through pip.
+
+Use the command from https://github.com/dmlc/gluon-nlp/blob/master/docs/README.txt to install the necessary packages.
+
+.. code:: bash
+
+    pip install sphinx>=1.5.5 sphinx-gallery sphinx_rtd_theme matplotlib Image recommonmark
```

Comment:
Isn't this covered by `pip install --user -e '.[extras,dev]'`? If so, let's delete it. If not, let's fix `setup.py` to cover it.